veps/sig-compute/vmtemplate-proposal.md

```diff
@@ -0,0 +1,733 @@
+To better keep track of the resources associated with a `VirtualMachineTemplate`
+the status of the CRD could be extended with a `Ready` condition and a list of
+volumes on which the template is dependent. The list should also contain the
+`Ready` condition of the volumes.
+
+This function requires that the `VirtualMachineTemplate` controller evaluates
+all `DataVolumeTemplates` in a specific template and constructs the status from
+all volumes that it can resolve.
+
+#### Import/Export of VirtualMachineTemplates in standardized format
+
+As previously mentioned, to make VM templates reusable, an import and export
+mechanism is required. Ideally, this mechanism should use a common and
+standardized format, to not only allow the import of templates created by
+KubeVirt, but also the import of appliances created by software vendors and
+templates created through other tools. As a widely adopted open standard in the
+virtualization industry, OVA/OVF is the ideal format for KubeVirt to implement.
+
+The entry point for the import mechanism should be a new
+`VirtualMachineTemplateImportRequest `CRD, which contains the required fields to
+specify a remote location where an OVA/OVF file can be obtained from and where
+it should be imported to. Another option could be the direct upload into the
+cluster through `virtctl` and possibly with the help of CDI.
+
+Once the OVA/OVF file can be processed, the contained disks need to be imported
+into PVCs and the attached virtual machine definition needs to be converted into
+a KubeVirt `VirtualMachine`. Once these tasks are completed the controller can
+create a new `VirtualMachineTemplate` from the results. This new template can
+then be used to create actual `VirtualMachines` from the imported template.
+
+Similar to the import mechanism, the entry point to the export mechanism should
+be a new `VirtualMachineTemplateExportRequest` CRD containing the required
+fields to specify either an existing `VirtualMachineTemplate` to be exported or
+an existing `VirtualMachine` to be exported directly. Once the exported template
+is ready to be downloaded, the controller should make the URL to download the
+template available in the status of the new CRD.
+
+For both import and export, `virtctl` should be extended to enable the feature
+to be used easily via the CLI.
+
+## API Examples
+
+For API examples, see the code blocks in the `template.kubevirt.io/v1alpha1`
+design.
+
+## Alternatives
+
+### Develop feature inside the core KubeVirt repository
+
+The current design proposes to develop this feature outside the KubeVirt core
+repository. An alternative would be to develop the feature within the KubeVirt
+core repository. However, this would lead to a slowdown in development as it
+would further increase the burden on reviewers in the core repository, would
+require all KubeVirt e2e tests to be run, most of which have nothing to do with
+the feature, and it would further increase the already substantial codebase.
+
```

Comment:
One positive is that we'd avoid the maintenance burden that another operator brings but IMHO in this case that's worth it given what you've said above.